src/Core/Framework/DataAbstractionLayer/Exception/AssociationNotFoundException.php

```diff
@@ -5,6 +5,9 @@
 use Shopware\Core\Framework\Log\Package;
 use Shopware\Core\Framework\ShopwareHttpException;
 
+/**
+ * @deprecated tag:v6.8.0 - reason:remove-exception - Will be removed, use {DomainException}::associationNotFound() instead
+ */
 #[Package('framework')]
 class AssociationNotFoundException extends ShopwareHttpException
```

Comment:
For an easier transition, these exceptions should start with the domain exception and inherit the error code